server.js

```diff
@@ -1,30 +1,111 @@
 import express from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
+import booksData from './data/books.json'
+
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
 //   PORT=9000 npm start
 const port = process.env.PORT || 8080
 const app = express()
 
+
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
+// Start defining your routes here. Below: 
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('Possible routes: /books, /language/:language, /rating, /page')
+})
+
+//SHOW WELCOME 
+app.get('/Welcome', (req, res) => {
+  res.send('WELCOME')
+})
+
+
+//SHOW A BOOK BASED ON ID
+app.get('/books/:id', (req, res) => {
+  const bookId = req.params.id
+  const book = booksData.find((item) => item.bookID === +bookId)
+
+  if (book) {
+    res.json(book);
+  } else {
+    res.status(404).send(`No book found with id ${bookId}.`);
+  }
+})
+
+// Filter based on title AND/OR author. http://localhost:8080/books?title=harry
+app.get("/books", (req, res) => {
+  //Query parameter
+  const titleSearch = req.query.title
+  const authorsSearch = req.query.authors
+  let filteredBooks = booksData
+
+  if (titleSearch) {
+    filteredBooks = filteredBooks.filter(book => {
+      const bookTitle = book.title.toString()
+      return bookTitle.toLowerCase().includes(titleSearch.toLowerCase())
+    })
+  }
+  if (authorsSearch) {
+    filteredBooks = filteredBooks.filter(book => {
+      const authorsName = book.authors.toString()
+      return authorsName.toLowerCase().includes(authorsSearch.toLowerCase())
+    })
+    //if array is empty show error message
+  } if (filteredBooks.length === 0) {
+    res.status(404).send(`No books found`);
+  } else {
+    res.json(filteredBooks)
+  }
+})
+
+//Filters on language_code.  http://localhost:8080/language/eng
+app.get('/language/:language', (req, res) => {
```

Comment:
It would be nice if this was part of the `/books` endpoint (and it'd be more RESTful) - then make use of query params to filter on language. So you could have `/books` to return all books and then `/books?language=en` for english books, etc.